### ISSUE TITLE
HomeKit does not update thermostat state

### ISSUE THREAD

**Issue by Denny5667** [arachnetech/homebridge-mqttthing#270]
i have faced an issue with thermostat. Unfortunately, HomeKit does not update it is state when the device update getCurrentHeatingCoolingState variable. Could you please check it? Thank you in advance.

**Comment by tobekas** [comment]
Could you post the content of your `config.json`?

**Comment by Denny5667** [comment]
Sure. Here is the config:

{
    "bridge": {
        "name": "HomeBridge",
        "username": "C8:67:EB:44:DF:64",
        "port": 28888,
        "pin": "327-12-016"
    },
    "platforms": [
        {
            "name": "Config",
            "port": 8080,
            "auth": "form",
            "theme": "dark-mode",
            "restart": "sudo -n systemctl restart homebridge",
            "temp": "/sys/class/thermal/thermal_zone0/temp",
            "tempUnits": "c",
            "lang": "auto",
            "sudo": true,
            "log": {
                "method": "systemd",
                "service": "homebridge"
            },
            "platform": "config"
        }
    ],
    "accessories": [
        {
            "accessory": "mqttthing",
            "type": "thermostat",
            "name": "ESP386806",
            "url": "mqtt://127.0.0.1",
            "username": "user",
            "password": "password",
            "topics": {
                "getCurrentHeatingCoolingState": "/ESP386806/Thermostat/1/getCurrentHeatingCoolingState",
                "setTargetHeatingCoolingState": "/ESP386806/Thermostat/1/setTargetHeatingCoolingState",
                "getCurrentTemperature": "/ESP386806/Thermostat/1/getCurrentTemperature",
                "setTargetTemperature": "/ESP386806/Thermostat/1/setTargetTemperature",
                "getTargetTemperature": "/ESP386806/Thermostat/1/getTargetTemperature",
                "getCurrentRelativeHumidity": "/ESP386806/Thermostat/1/getCurrentRelativeHumidity"
            },
            "heatingCoolingStateValues": [
                "0",
                "1"
            ],
            "minTemperature": 18,
            "maxTemperature": 28,
            "restrictHeatingCoolingState": [
                0,
                1
            ]
        }
      ]
}

**Comment by rbswift** [comment]
what output do you get from this command?

`mosquitto_sub -h localhost -t '/ESP386806/Thermostat/1/getCurrentHeatingCoolingState' -v`

Is the leading '/' before ESP386806 an error?

`mosquitto_sub -h localhost -t 'ESP386806/Thermostat/1/getCurrentHeatingCoolingState' -v`

**Comment by Denny5667** [comment]
No, it is not a mistake and slash does not cause any problems. The telegrams are being received correctly. I already use many configs and never got such problem. Even by the thermostat config, if the device post target temperature in the getTargetTemperature topic, the HomeKit receives it and shows updated target temperature correctly. But if the device tells HomeKit, that it has been switched on or off, or it is power status after reboot, for example, the HomeKit does not change the state of thermostat.

**Comment by Denny5667** [comment]
Thanks for advice. I habe noticed, that i use different topic in my code as in the config above.
My bad guys, sorry for inconvenience.

**Comment by Denny5667** [comment]
I have discovered another issues. HomeKit has very strange behaviour...
So, after i have corrected the code, the HomeKit change the status of icon, but not correctly. If the thermostat is disabled in the HomeKit and i update it's state by sending in getCurrentHeatingCooling "1", the HomeKit highlight the icon of thermostat, change the state in the notification area to "heating", but the temperature indicator remains grey. If i open controls, the switch button is also remains grey. Now, i press the switch button in the HomeKit and the temperature icon becomes orange. After that, i send in the getCurrentHeatingCoolingState "0" and the temperature icon changes it's color to green, but the HomeKit does not change thermostat state. Is still remains "heating" and the switch button remains orange...

**Comment by rbswift** [comment]
Check if you have missing configuration for any topics that are not marked optional.

Specifically, make sure you have mqtt topics configured for both getCurrentHeatingCoolingState and get**Target**HeatingCoolingState. These topics can be the same.

**Comment by arachnetech** [comment]
I agree - sounds like you need to update the target state in addition to the current state.
